js/web/lib/wasm/jsep/backend-webnn.ts

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// Current version of the WebNN API does not have a TypeScript definition file.
+// https://github.com/webmachinelearning/webnn/issues/677
+/// <reference path="./webnn/webnn.d.ts" />
+
+
+class ContextState {
+    context: MLContext;
+    sessions: Set<number>;
+}
+
+export class WebNNBackend {
+    contextStates: Map<string, ContextState>
+    contextsBySessionId: Map<number, string>
+
+    constructor() {
+        this.contextStates = new Map();
+        this.contextsBySessionId = new Map();
+    }
+
+    async createContext(options: MLContextOptions): Promise<string> {
+        const contextId = JSON.stringify(options);
+        if(this.contextStates.has(contextId)) {
+            return contextId;
+        }
+        const context = await navigator.ml.createContext(options);
```

Comment:
For `MLPowerPreference`, currently [WebNN implementation](https://source.chromium.org/chromium/chromium/src/+/main:third_party/blink/renderer/modules/ml/ml_context_options.idl;l=29?q=ml_context_options) reuses the context options from Model Loader API, which uses `auto` rather then `default` in WebNN spec. 

If we pass `default` to `MLContextOptions`, it will throw `'default' is not a valid enum value of type MLPowerPreference`.

So we should make a workaround here, if `options.powerPreference` is `'default'`, just ignore this option.

Reply:
I added a check to ignore `default`.